Stride.BepuPhysics/Components/Colliders/ConvexHullColliderComponent.cs

```diff
@@ -1,19 +1,74 @@
-﻿using Stride.BepuPhysics.Processors;
+﻿using BepuPhysics;
+using BepuPhysics.Collidables;
+using BepuUtilities.Memory;
+using Stride.BepuPhysics.Extensions;
+using Stride.BepuPhysics.Processors;
 using Stride.Core;
 using Stride.Engine;
 using Stride.Engine.Design;
+using Stride.Games;
+using Stride.Physics;
 
 namespace Stride.BepuPhysics.Components.Colliders
 {
     [DataContract]
     [DefaultEntityComponentProcessor(typeof(ColliderProcessor), ExecutionMode = ExecutionMode.Runtime)]
     [ComponentCategory("Bepu - Colliders")]
-    public class ConvexHullColliderComponent : ColliderComponent
+    public sealed class ConvexHullColliderComponent : ColliderComponent
     {
-        public ModelComponent? ModelData { get; set; }
+        #warning Replace with an explicit reference to hulls once the asset part for hulls is done
+        public PhysicsColliderShape? Hull;
 
         public ConvexHullColliderComponent()
         {
         }
+
+        internal override void AddToCompoundBuilder(IGame game, ref CompoundBuilder builder, RigidPose localPose)
+        {
+            builder.Add(new ConvexHull(GetMeshPoints(), new BufferPool(), out _), localPose, Mass);
+        }
+
+        Span<System.Numerics.Vector3> GetMeshPoints()
+        {
+            if (Hull == null)
+                return new();
+
+            int vertCount = 0;
+            foreach (var colliderShapeDesc in Hull.Descriptions)
+            {
+                if (colliderShapeDesc is ConvexHullColliderShapeDesc hullDesc) // This casting nonsense should be replaced once we have a proper asset to host convex shapes
```

Comment:
Im guessing this is required due to the compound shape possibly having multiple types of shapes?

Reply:
Yeah, something we'll have to change in engine, defining a single explicit asset file just for hulls